lib/package.py

```diff
@@ -101,8 +101,46 @@ def __init__(self, name, force_rebuild=True):
             raise exception.PackageDescriptorError(
                 "Failed to find %s's YAML descriptor" % self.name)
 
+        # load package metadata YAML file
         self._load()
 
+        # get global config information which may override package YAML
+        # Package option fields are separated by `#` character and the expected
+        # format can be one of the following:
+        #
+        #   package_name#repo_url#branch_name#revision_id
+        #   package_name#repo_url#reference
+        #   package_name##reference
+        #   package_name
+        packages_options = CONF.get('build_packages').get('packages') or []
+        for package_option in packages_options:
+            package_parts = package_option.split("#")
+            if package_parts[0] == name:
+                # cancel if there are no sources
+                if not self.sources:
+                    break
+                # assume that the first source is the main one and that a source
+                # override targets it
+                source_type = self.sources[0].keys()[0]
+                main_source = self.sources[0][source_type]
+                if len(package_parts) == 1:
+                    pass
+                elif len(package_parts) == 2:
+                    if package_parts[1]:
+                        main_source["src"] = package_parts[1]
```

Comment:
I still need to test if len == 2, if len >4 or something similar to be able to infer if number of parts is correct. IMHO testing each length is more readable.